src/mod/endpoints/mod_sofia/sofia_presence.c

```diff
@@ -2987,12 +2987,27 @@ static int sofia_presence_sub_callback(void *pArg, int argc, char **argv, char *
 
 			if (!strcasecmp(astate, "early") || !strcasecmp(astate, "confirmed")) {
 
+				char *cmd;
+				char callee_id_name[16] = { 0 };
+				switch_stream_handle_t user_data_stream = { 0 };
+
 				clean_to_user = switch_mprintf("%s", sub_to_user ? sub_to_user : to_user);
 				clean_from_user = switch_mprintf("%s", from_id ? from_id : from_user);
 
+				SWITCH_STANDARD_STREAM(user_data_stream);
+				cmd = switch_mprintf("%s@%s var effective_caller_id_name", clean_to_user, host);
+				if (switch_api_execute("user_data", cmd, NULL, &user_data_stream) == SWITCH_STATUS_SUCCESS) {
+					if (strncmp("-ERR", user_data_stream.data, 4)) {
+						switch_snprintf(callee_id_name, sizeof(callee_id_name), "%s", (char *) user_data_stream.data);
```

Comment:
do you really need to copy?

Reply:
This looked cleaner to my eye but I will revert as you requested earlier.

Reply:
Without me reverting, you can see that commit here:
https://github.com/signalwire/freeswitch/commit/8ea0af693294d675fb494c5c923853f0389b2c7c

Reply:
The question above remains.

Reply:
switch_safe_free(callee_id_name); on 3083 doesn't help that?

Reply:
the question is do you really need to copy?

Reply:
Sorry I'm not getting it and I'm sure it's on my end :)

You're saying even with `switch_mprintf` there's another way to do this by directly referring to `user_stream_data.stream` below?  What if I return -ERR?

I could refactor the whole thing to be in the `if (is_dialog) `branch and pass directly to stream.write_function... Is that what you're suggesting?

I appreciate your time and patience.

Reply:
You can set `callee_id_name` to `user_data_stream.data` when you need and forget about freeing `callee_id_name` as you free the stream anyways.

Reply:
I thought it's a memory leak to reassign a pointer to another pointer since the initial memory will remain unfreed.  Looking at it now, user_data_stream isn't a pointer...

OK, I pushed that.  I left out the typecast of (char *)user_data_stream because I don't know how it's possible to point to a cast (beyond my pay grade).

Reply:
you need to cast on the right.

Reply:
I am reverting back to copying.
I noted below that we potentially exceed MTU in UDP packets.  Since CallerID is only supposed to contain 15 characters, the string should be truncated.  We copy with switch_snprintf and truncate to 16 bytes.  This will help alleviate at least some of the problem.